src/PokemonSuper.java

```diff
@@ -0,0 +1,140 @@
+package src;
+
+public abstract class PokemonSuper {
+    final String name;
+    String nameOfTrainer;
+    int level;
+    private int healthPoints;
+    private int experiencePoints;
+
+    String pokemonTalk;
+    String favouriteFood;
+    String defence;
+    String attack;
+
+    boolean beingAttacked;
+
+    FirePokemon firePokemon;
+    WaterPokemon waterPokemon;
+    GrassPokemon grassPokemon;
+    ElectricPokemon electricPokemon;
+
+    public FirePokemon getFirePokemon() {
+        return firePokemon;
+    }
+
+    public void setFirePokemon(FirePokemon firePokemon) {
+        this.firePokemon = firePokemon;
+    }
+
+    public WaterPokemon getWaterPokemon() {
+        return waterPokemon;
+    }
+
+    public WaterPokemon setWaterPokemon(WaterPokemon waterPokemon) {
+        return this.waterPokemon = waterPokemon;
+    }
+
+    public GrassPokemon getGrassPokemon() {
+        return grassPokemon;
+    }
+
+    public GrassPokemon setGrassPokemon(GrassPokemon grassPokemon) {
+        return this.grassPokemon = grassPokemon;
+    }
+
+    public ElectricPokemon getElectricPokemon() {
+        return electricPokemon;
+    }
+
+    public void setElectricPokemon(ElectricPokemon electricPokemon) {
+        this.electricPokemon = electricPokemon;
+    }
+
+    public PokemonSuper(String name, String nameOfTrainer, int level, int healthPoints, int experiencePoints, String pokemonTalk, String favouriteFood, String defence, String attack, boolean beingAttacked) {
```

Comment:
Je hebt je constructor nu tussen je getters/setters staan. Conventie is om je klasse als volgt te verdelen: 

attributen

constructor(s)

getters/setters

andere methodes